File: frontend/src/api/recipe.ts
import { request } from '@/utils/request'
import type { PageResult } from '@/types/api'

export interface RecipeItem {
  id: number
  recipeName: string
  category?: string
  coverImage?: string
  summary?: string
  efficacy?: string
  cookingSteps?: string
  cookingTime?: string
  difficulty?: string
  tags?: string
  constitutionTags?: string
  isFeatured?: number
  viewCount?: number
}

export function fetchRecipeFeatured() {
  return request<RecipeItem>({
    url: '/recipes/featured',
    method: 'get',
  })
}

export function fetchRecipePage(params: {
  page?: number
  pageSize?: number
  category?: string
}) {
  return request<PageResult<RecipeItem>>({
    url: '/recipes',
    method: 'get',
    params,
  })
}

export function fetchRecipeDetail(id: number) {
  return request<RecipeItem>({
    url: `/recipes/${id}`,
    method: 'get',
  })
}

File: backend/src/main/java/com/bencao/entity/Recipe.java
package com.bencao.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("recipe")
public class Recipe {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String recipeName;

    private String category;

    private String coverImage;

    private String summary;

    private String efficacy;

    private String cookingSteps;

    private String cookingTime;

    private String difficulty;

    private String tags;

    private String constitutionTags;

    private Integer isFeatured;

    private Integer viewCount;

    private Integer status;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}
